examples/simple_mdns/simple_mdns_device.c

```diff
@@ -0,0 +1,83 @@
+//
+// Simple example to demonstrate a Nabto Edge mDNS enabled device. Combine with other examples to
+// achieve anything in addition to just service discovery. See the Nabto Edge tunnel app for a full
+// example.
+//
+
+#include <nabto/nabto_device.h>
+#include <apps/common/string_file.h>
+
+#ifdef _WIN32
+#include <Windows.h>
+#define NEWLINE "\r\n"
+#else
+#include <unistd.h>
+#define NEWLINE "\n"
+#endif
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <signal.h>
+#include <stdlib.h>
+
+void die(const char* msg) {
+    printf("%s" NEWLINE, msg);
+    exit(1);
+}
+
+int main(int argc, char* argv[]) {
+
+    if (argc != 6) {
+        printf("The example takes exactly 5 arguments: %s <product-id> <device-id> <mdns subtype> <mdns txt key> <mdns txt value>" NEWLINE, argv[0]);
+        return -1;
+    }
+
+    char* productId = argv[1];
+    char* deviceId = argv[2];
+    char* subType = argv[3];
+    char* txtKey = argv[4];
+    char* txtVal = argv[5];
+
+    NabtoDevice* device;
+
+    printf("Nabto Embedded SDK Version %s\n", nabto_device_version());
+
+    if ((device = nabto_device_new()) == NULL) {
+        die("Allocation error");
+    }
+
+    char *key;
+    if (nabto_device_create_private_key(device, &key) != NABTO_DEVICE_EC_OK ||
```

Comment:
set_private_key() copies the key into internal storage, so you must free your private key yourself using nabto_device_string_free().

sidenote: It may be irrelevant for this example, however, creating a new key every time and not printing it means the device will never be able to attach to the bs, and you will keep seeing the log line stating this. This may also be irrelevant for this example since you are using log level trace.

Reply:
Key is now freed. Attach is not relevant for the example and will just complicate unnecessarily IMHO (see intro comment about purpose and disclaimar).